Source/Scene/ClippingPlanesCollection.js

```diff
@@ -0,0 +1,245 @@
+define([
+        '../Core/Cartesian3',
+        '../Core/Cartesian4',
+        '../Core/Check',
+        '../Core/Color',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/Intersect',
+        '../Core/Matrix4',
+        '../Core/Plane'
+    ], function(
+        Cartesian3,
+        Cartesian4,
+        Check,
+        Color,
+        defaultValue,
+        defined,
+        defineProperties,
+        Intersect,
+        Matrix4,
+        Plane) {
+    'use strict';
+
+    /**
+     * Specifies a set of clipping planes. Clipping planes selectively disable rendering in a region on the outside of the specified list of {@link Plane} objects.
+     *
+     * @alias ClippingPlanesCollection
+     * @constructor
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Plane[]} [options.planes=[]] An array of up to 6 {@link Plane} objects used to selectively disable rendering on the outside of each plane.
+     * @param {Boolean} [options.enabled=true] Determines whether the clipping planes are active.
+     * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix specifying an additional transform relative to the clipping planes original coordinate system.
+     * @param {Boolean} [options.combineClippingRegions=true] If true, the region to be clipped must be included in all planes in this collection. Otherwise, a region will be clipped if included in any plane in the collection.
+     * @param {Color} [options.edgeColor=Color.WHITE] The color applied to highlight the edge along which an object is clipped.
+     * @param {Number} [options.edgeWidth=0.0] The width, in pixels, of the highlight applied to the edge along which an object is clipped.
+     */
+    function ClippingPlanesCollection(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * An array of up to 6 {@link Plane} objects used to selectively disable rendering on the outside of each plane.
```

Comment:
Why 6?  What happens if you add more?  Should we throw a developer error when someone tries to do that?

Reply:
This is an arbitrary limit we have based on the uniform we pass to the shader. We'd like to include more, but need to restructure how we pass in the planes, see the [roadmap](https://github.com/AnalyticalGraphicsInc/cesium/issues/5765#issuecomment-335950848). If set to more, they are simply ignored.

Reply:
We have it at 6 for now because that's all the faces of a cube.

Reply:
Okay, that makes sense.  I think we want to have this class follow the same API as the other `XXXCollection` classes, so make `this.planes` -> `this._planes`, then in the constructor and the `ClippingPlanesCollection.prototype.add` function throw a developer error if the length is greater than 6.  That way people using clipping planes will be aware of what's going on instead of wondering why their planes were ignored